### PR TITLE
Fixe for issue #266, #265 and refactoring

### PR DESCRIPTION
**Pull request by akharit**
---

This checklist is used to make sure that common guidelines for a pull request are followed.

### Description of the change
Fixe for issue #265 and #265. Also some refactoring for multiprocessor acl change code.
### General Guidelines

- [ ] The PR has modified HISTORY.rst with an appropriate description of the change and a version increment.
Will be done when merge to master later and making a new release
- [x] The PR has supporting test coverage that confirm the expected behavior and protects against regressions, including necessary recordings.
- [x] Links to associated bugs, if any, are in the description.

### CODE REVIEW

File: azure/datalake/store/core.py [hunk 1]
```diff
@@ -865,6 +865,8 @@ def read(self, length=-1):
             data_read = self.cache[self.loc - self.start:
                              min(self.loc - self.start + length, self.end - self.start)]
             out += data_read
+            if len(data_read) == 0: # Check to catch possible server errors. Ideally shouldn't happen.
```
Comment (lmazuel) [inline azure/datalake/store/core.py]: For what I understand, `data_read` is an array. Then the recommended way to test if it's empty is `if not data_read`:
https://stackoverflow.com/questions/43121340/why-is-the-use-of-lensequence-in-condition-values-considered-incorrect-by-pyli

File: setup.py [hunk 2]
```diff
@@ -5,11 +5,17 @@
 from io import open
 import re
 try:
-    from azure_bdist_wheel import cmdclass
+    import azure
+    try:
+        ver = azure.__version__
```
Comment (lmazuel) [inline setup.py]: This try except logic was to test "azure" package as it was 3.5 years ago. I think it's safe for you to not introduce it and keep you file simple.

**Comment by lmazuel** [comment]
LGTM to me, thanks for asking me a reviewer!
To double-check everything works:
- Your sdist must contains `azure/__init__.py` (with the pkgutil syntax)
- Your wheel must NOT contain `azure/__init__.py`
- Install the wheel on Python 3 must NOT install azure-nspkg

File: azure/datalake/store/core.py (hunk 1 above)
Comment (akharit) [reply azure/datalake/store/core.py]: Changed it to not(data_read)

File: setup.py (hunk 2 above)
Comment (akharit) [reply setup.py]: Can you confirm the changes I pushed? We don't need to import azure package, right?

Comment (lmazuel) [reply setup.py]: Yes we don't, changes looks good to me

File: setup.py [hunk 3]
```diff
@@ -44,17 +37,21 @@
...
+      packages=find_packages(exclude=['tests',
+                                      # Exclude packages that will be covered by PEP420 or nspkg
+                                      'azure',
+                                      ]),
       install_requires=[
           'cffi',
           'adal>=0.4.2',
-          'requests>=2.20.0'
+          'requests>=2.20.0',
+          'azure-common~=1.1'
```
Comment (lmazuel) [inline setup.py]: You're not required to do that if you don't import "azure.common"

File: setup.py [hunk 4]
```diff
@@ -44,17 +37,21 @@
...
       ],
       extras_require={
           ":python_version<'3.4'": ['pathlib2'],
           ":python_version<='2.7'": ['futures'],
+          ":python_version<'3.0'": ['azure-mgmt-nspkg'],
```
Comment (lmazuel) [inline setup.py]: Jsut realized I read too fast the first time, should be "azure-nspkg"

**Pull request merged**